Update transactions doc to include bulk delete and timeouts (#240)
xataio/mdx-docs: 040-SDK/104-delete.mdx

</TabbedCode>

> To delete multiple records we recommend using transactions. [Transactions](/docs/sdk/transaction#delete) will not fail a delete operation if a record is not found. All operations in a transaction must succeed otherwise it is rolled back.

Reviewer: ```suggestion
To delete multiple records from the API we recommend using transactions. [Transactions](/docs/sdk/transaction#delete) will not fail a delete operation if a record is not found. All operations in a transaction must succeed otherwise it is rolled back.
```

> To avoid the SQL proxy's 60-second timeout, delete in small batches. We recommend deleting no more than 10,000 records per batch. You can use a nested `SELECT` statement to manage batch sizes efficiently and prevent timeouts. For example, as in the following:

Reviewer: ```suggestion
Alternatively you can use the `DELETE` SQL statement via the [SQL Proxy](/docs/sdk/sql/overview), but note that there is an implicit timeout of 60 seconds which may be reached when processing many records. To avoid reaching the SQL proxy's timeout, limit the delete statement to smaller batches. You can use a nested `SELECT` statement to control the batch size. For example, as in the following:
```

```sql

> DELETE FROM titles

Reviewer: ```suggestion
DELETE FROM "titles"
```

WHERE id IN (
    SELECT id

>     FROM titles

Reviewer: ```suggestion
    FROM "titles"
```

>     WHERE originalTitle ILIKE '%Matrix%'

Reviewer: ```suggestion
    WHERE "originalTitle" ILIKE '%Matrix%'
```

>     LIMIT 10000

Reviewer: ```suggestion
    ORDER BY "xata.createdAt" ASC
    LIMIT 10000
```

);
```

- `SELECT id`: Specifies that only the `id` column will be retrieved in the query.

> - `FROM titles`: The subquery operates on the `titles` table.

Reviewer: ```suggestion
- `FROM "titles"`: The subquery operates on the `titles` table.
```

> - `WHERE originalTitle ILIKE '%Matrix%'`: Filters rows based on a case-insensitive partial match in the `originalTitle` column.

Reviewer: ```suggestion
- `WHERE "originalTitle" ILIKE '%Matrix%'`: Filters rows based on a case-insensitive partial match in the `originalTitle` column.
```
